File: CS-Examples/02_Data/CopyDataWithStyle.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.IO;
using Spire.Xls;
using Spire.Xls.Charts;

namespace CopyDataWithStyle
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a workbook
			Workbook workbook = new Workbook();
      
            //Get the default first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            //Set the values for some cells.
            CellRange cells = worksheet.Range["A1:J50"];
            for (int i = 1; i <= 10; i++)
            {
                for (int j = 1; j <= 8; j++)
                {
                    string text = string.Format((i - 1).ToString() + "," + (j - 1).ToString());
                    cells[i, j].Text = text;
                }
            }
            //Get a source range (A1:D3).
            CellRange srcRange = worksheet.Range["A1:D3"];

            //Create a style object.
            CellStyle style = workbook.Styles.Add("style");

            //Specify the font attribute.
            style.Font.FontName = "Calibri";

            //Specify the shading color.
            style.Font.Color = Color.Red;

            //Specify the border attributes.
            style.Borders[BordersLineType.EdgeTop].LineStyle = LineStyleType.Thin;
            style.Borders[BordersLineType.EdgeTop].Color = Color.Blue;
            style.Borders[BordersLineType.EdgeBottom].LineStyle = LineStyleType.Thin;
            style.Borders[BordersLineType.EdgeBottom].Color = Color.Blue;
            style.Borders[BordersLineType.EdgeTop].LineStyle = LineStyleType.Thin;
            style.Borders[BordersLineType.EdgeTop].Color = Color.Blue;
            style.Borders[BordersLineType.EdgeRight].LineStyle = LineStyleType.Thin;
            style.Borders[BordersLineType.EdgeRight].Color = Color.Blue;
            srcRange.CellStyleName = style.Name;

            //Set the destination range
            CellRange destRange = worksheet.Range["A12:D14"];

            //Copy the range data with style
            srcRange.Copy(destRange, true, true);
     
            //String for output file 
            String outputFile = "Output.xlsx";

            //Save the file
            workbook.SaveToFile(outputFile, ExcelVersion.Version2013);

            //Launching the output file.
            Viewer(outputFile);
		}
		private void Viewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: VB-Examples/02_Data/CopyDataWithStyle.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO
Imports Spire.Xls
Imports Spire.Xls.Charts

Namespace CopyDataWithStyle
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object.
            Dim workbook As New Workbook()

            ' Get the first worksheet from the workbook.
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Set values for cells in the range A1 to J50 using nested loops.
            Dim cells As CellRange = worksheet.Range("A1:J50")
            For i As Integer = 1 To 10
                For j As Integer = 1 To 8
                    Dim text As String = String.Format((i - 1).ToString() & "," & (j - 1).ToString())
                    cells(i, j).Text = text
                Next j
            Next i
            ' Get a source range (A1:D3) from the worksheet.
            Dim srcRange As CellRange = worksheet.Range("A1:D3")

            ' Create a new cell style and specify font, color, and border attributes for the style.
            Dim style As CellStyle = workbook.Styles.Add("style")

            style.Font.FontName = "Calibri"
            style.Font.Color = Color.Red
            style.Borders(BordersLineType.EdgeTop).LineStyle = LineStyleType.Thin
            style.Borders(BordersLineType.EdgeTop).Color = Color.Blue
            style.Borders(BordersLineType.EdgeBottom).LineStyle = LineStyleType.Thin
            style.Borders(BordersLineType.EdgeBottom).Color = Color.Blue
            style.Borders(BordersLineType.EdgeTop).LineStyle = LineStyleType.Thin
            style.Borders(BordersLineType.EdgeTop).Color = Color.Blue
            style.Borders(BordersLineType.EdgeRight).LineStyle = LineStyleType.Thin
            style.Borders(BordersLineType.EdgeRight).Color = Color.Blue

            ' Apply the created style to the source range.
            srcRange.CellStyleName = style.Name

            ' Set the destination range in the worksheet.
            Dim destRange As CellRange = worksheet.Range("A12:D14")

            ' Copy the data from the source range to the destination range while preserving the style.
            srcRange.Copy(destRange, True, True)

            ' Define a string variable named outputFile to store the output file name as "Output.xlsx".
            Dim outputFile As String = "Output.xlsx"

            ' Save the modified workbook to a file with the name specified in outputFile, using Excel 2013 format.
            workbook.SaveToFile(outputFile, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launching the output file.
            Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
